inference/core/env.py

```diff
@@ -283,6 +283,8 @@
 
 # Maximum embedding cache size for SAM, default is 10
 SAM_MAX_EMBEDDING_CACHE_SIZE = int(os.getenv("SAM_MAX_EMBEDDING_CACHE_SIZE", 10))
+# The sam2 low_res_masks are the biggest memory usage, and 1000 of them take 256*256*4*1000/1024/1024 MB = 250MB
+SAM2_MAX_CACHE_SIZE = int(os.getenv("SAM_MAX_EMBEDDING_CACHE_SIZE", 1000))
```

Comment:
I propose to decouple SAM from SAM2 in terms of env variables configuring model
plus let's enlist the config variable in [this page](https://inference.roboflow.com/server_configuration/environmental_variables/)